### PR TITLE
Update SETUP.md

### PR DESCRIPTION
**Pull request by jamescodeorg**
Updating setup.md with experience gained from me and cforkish setting up our dev environments.

Notable changes:
* Updated Mac OS instructions for OS X Catalina
* Fixed formatting for <detail> sections by unindenting the markdown so it wouldn't be treated as code blocks

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: SETUP.md [hunk 1]
```diff
@@ -18,20 +18,22 @@ You can do Code.org development using OSX, Ubuntu, or Windows (running Ubuntu in
 1. `gem install bundler -v 1.17`
 1. `rbenv rehash`
 1. `cd code-dot-org`
-1. `bundle install` (Problems running this step? See [tips](#bundle-install-tips) below.)
+1. `bundle install`
+    * This step often fails to due environment-specific issues. Look in the [Bundle Install Tips](#bundle-install-tips) section below for steps to resolve many common issues.
 1. `bundle exec rake install:hooks`
     <details>
-        <summary>Troubleshoot: `rake aborted!..` </summary>
+      <summary>Troubleshoot: `rake aborted!..` </summary>
```
Comment (davidsbailey) [inline SETUP.md]: "rake aborted" is likely to appear in almost any failure here. I would suggesting including at least `rake aborted! Gem::LoadError...`

File: SETUP.md [hunk 2]
```diff
@@ -49,20 +51,52 @@ After setup, read about our [code styleguide](./STYLEGUIDE.md), our [test suites
 
 ## OS-specific prerequisites
 
-### OS X Mojave / Mavericks / Yosemite / El Capitan / Sierra
+### OS X Catalina
 
+1. Choose shell. Starting in Catalina, [the default shell for new users is zsh](https://support.apple.com/en-us/HT208050). Most developers at Code.org are still using bash so that may be a smoother experience for now.
```
Comment (davidsbailey) [inline SETUP.md:56]: Fair. :-)

File: SETUP.md [hunk 3]
```diff
@@ -49,20 +51,52 @@ After setup, read about our [code styleguide](./STYLEGUIDE.md), our [test suites
...
 1. Install Redis: `brew install redis`
 1. Run `brew install https://raw.github.com/quantiverge/homebrew-binary/pdftk/pdftk.rb enscript gs mysql@5.7 nvm imagemagick rbenv ruby-build coreutils sqlite parallel`
     <details>
-        <summary>Troubleshoot: <code>Formula.sha1</code> is disabled or <code>Error: pdftk: undefined method sha1' for #&lt;Class:...&gt;</code></summary>
+      <summary>Troubleshoot: pdftk errors</summary>
 
-        If it complains about `Formula.sha1` is disabled, removing https://raw.github.com/quantiverge/homebrew-binary/pdftk/pdftk.rb from the above command seems to not have serious side effects (it will cause `PDFMergerTest` to fail). It may be a new URL is needed in the dependency list, see https://leancrew.com/all-this/2017/01/pdftk/
+      * If it complains about pdftk, removing https://raw.github.com/quantiverge/homebrew-binary/pdftk/pdftk.rb from the above command seems to not have serious side effects (it will cause `PDFMergerTest` to fail). It may be a new URL is needed in the dependency list, see https://leancrew.com/all-this/2017/01/pdftk/
```
Comment (davidsbailey) [inline SETUP.md:94]: I forgot to ask @mvkski before he left whether we're likely to keep using pdftk for a while, or if we maybe have a plan to move off of it. 

either way, the workaround here look good to me.

**Review by davidsbailey — APPROVED** [review]
Thanks for these updates, @jamescodeorg ! Looks good to me after addressing one request below.

**Review by cforkish — APPROVED** [review]
LGTM! there are a few things from my PR that i think might still be nice to include so i'll rebase that once this is merged.

**Pull request merged**